ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -648,35 +646,36 @@ private void performHardDeletes(List<MessageInfo> messageInfoList) throws StoreE
       Iterator<HardDeleteInfo> hardDeleteIterator = hardDelete.getHardDeleteMessages(readSet, factory, null);
       Iterator<BlobReadOptions> readOptionsIterator = readOptionsList.iterator();
 
-      /* Next, get the information to persist hard delete recovery info. Get all the information and save it, as only
-       * after the whole range is persisted can we start with the actual log write */
-      while (hardDeleteIterator.hasNext()) {
-        if (!enabled.get()) {
-          throw new StoreException("Aborting hard deletes as store is shutting down",
-              StoreErrorCodes.Store_Shutting_Down);
```

Comment:
any reason to remove this check?

Reply:
I actually don't have to, I will add it back